### PR TITLE
Split jobs doc

### PR DESCRIPTION
**Pull request by clementbolin**
## What does this PR do ?

Splitting of CI jobs into a new file.

**Why ?**
During a PR of a 2-dev branch we have seen this because the doc deployment jobs are skipped automatically because of this lines:
`if: github.event_name != 'pull_request'`
It also allows to have a more readable CI by not having all our jobs in the same file.

### CODE REVIEW

File: .github/workflows/workflow-doc.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+name: Workflow Doc
```
Comment (alexandrebouthinon) [inline .github/workflows/workflow-doc.yaml:1]: This workflow also deploy packages and Docker image maybe you could find a better name

Comment (clementbolin) [reply .github/workflows/workflow-doc.yaml:1]: Maybe something like: `Workflow deploy` or `Workflow deploy npm / docker / doc` I prefer the first option

Comment (alexandrebouthinon) [reply .github/workflows/workflow-doc.yaml:1]: I personally like "workflow deploy" or "Deployments"

**Review by Aschen — APPROVED** [review]
👍

**Pull request merged**